concepts/enums/about.md

```diff
@@ -131,6 +119,27 @@ enum Option<T> {
 }
 ```
 
-The Option enum is helpful because it allows you to explicitly represent the possibility of a value being absent, making your code more expressive and easier to reason about. Using Option can also help prevent bugs caused by using uninitialized or unexpected null values.
+The `Option` enum is helpful because it allows you to explicitly represent the possibility of a value being absent, making your code more expressive and easier to reason about. Using `Option` can also help prevent bugs caused by using uninitialized or unexpected `null` values.
+
+The function below shows how the `Option` enum is used to return the first elements of an array with a given value, or return `None` if the element is absent.
+
+```Rust
+fn find_value_iterative(mut arr: Span<felt252>, value: felt252) -> Option<usize> {
+    let mut result = Option::None;
+    let mut index = 0;
+
+    while let Option::Some(array_value) = arr
+        .pop_front() {
+            if (*array_value == value) {
+                result = Option::Some(index);
+                break;
+            };
+
+            index += 1;
+        };
+
+    result
+}
+```
 
-Other native enums such as `Result` enum, allows for thorough error handling. The Result enum is will explained in details in the `Error Handling` chapter.
+There are other native enums, one of which is the `Result` enum, which allows for graceful error handling. The `Result` enum will be explained in detail in the `Error Handling` chapter.
```

Comment:
```suggestion
There are other native enums, one of which is the `Result` enum, which allows for graceful error handling. Read more about the `Result` enum in [The Cairo Book](https://book.cairo-lang.org/ch09-02-recoverable-errors.html#the-result-enum).
```
I think it's actually better to just link to TCB, since we cannot link to said Exercism chapters. Also, we might launch an incomplete syllabus that doesn't have the "Error Handling" concept written, so this sentence would be misguiding to students.